packages/client/src/request-login.test.ts

```diff
@@ -1,11 +1,108 @@
-import { getLoginUrlAndCodeVerifier } from './request-login';
+import { DEFAULT_SCOPE_STRING } from './constants';
+import { getLoginUrlWithCodeVerifierAndState } from './request-login';
 
-test('getLoginUrlAndCodeVerifier', async () => {
-  const { url } = await getLoginUrlAndCodeVerifier({
-    baseUrl: 'http://logto.dev/oidc/auth',
-    clientId: 'foo',
-    scope: 'openid offline_access',
-    redirectUri: 'http://localhost:3000',
+const BASE_URL = 'http://logto.dev/oidc/auth';
+const CLIENT_ID = 'foo';
+const REDIRECT_URI = 'http://localhost:3000';
+
+describe('getLoginUrlAndCodeVerifier', () => {
+  test('url must start with baseUrl', async () => {
+    const { url } = await getLoginUrlWithCodeVerifierAndState({
+      baseUrl: BASE_URL,
+      clientId: CLIENT_ID,
+      scope: DEFAULT_SCOPE_STRING,
+      redirectUri: REDIRECT_URI,
+    });
+    expect(url.startsWith(BASE_URL)).toBeTruthy();
+  });
+
+  test('url must contain code_challenge=EXPECTED_VALUE', async () => {
+    const { url } = await getLoginUrlWithCodeVerifierAndState({
+      baseUrl: BASE_URL,
+      clientId: CLIENT_ID,
+      scope: DEFAULT_SCOPE_STRING,
+      redirectUri: REDIRECT_URI,
+    });
+    expect(url).toContain(`client_id=${CLIENT_ID}`);
+  });
+
+  test('url must contain scope=VALUE_FROM_PARAMETER', async () => {
```

Comment:
`VALUE_FROM_PARAMETER` style parameter is common used as constant and it's confusing in test description. 

maybe
```js
test('url must contains expected scope value')
```
would be better.

Reply:
Fixed.

Replaced by 'url must contain expected scope'.

- scope : a string contains multiple scope values, e.g. 'openid offline_access name'
- scope value : a single value e.g. 'openid', 'offline_access', 'name'